Client/weather-app/src/components/Weather.vue

```diff
@@ -0,0 +1,471 @@
+<template>
+<div>
+    <div class="row">
+        <div class="col-md-12">
+          <form style="margin:5px;">
+            <div class="form-row">
+              <div class="col-12 col-md-9 mb-2 mb-md-0">
+                <input
+                  type="text"
+                  v-model="inputValue"
+                  class="search"
+                  placeholder="Enter a city or German postcode..."
+                />
+              </div>
+              <div class="search-button-container">
+                <button
+                  type="button"
+                  class="search-button"
+                  v-on:click="searchWeather"
+                >
+                  Search
+                </button>
+              </div>
+
+               <div
+                v-if="showError"
+                class="error-message"
+            >
+                Invalid name or zipcode
+            </div>
+
+            </div>
+          </form>
+        </div>
+      </div>
+        <div class="weather-container">
+            <div
+                v-for="(weather, index) in weatherForecast"
+                :key="weather.date"
+            >
+                <div class="weather">
+                    <div class="current">
+                    <div class="info">
+                        <div>&nbsp;</div>
+                        <div class="city">
+                        <small>
+                            <small>CITY:</small>
+                        </small>
+                        {{ weather.city }}
+                        </div>
+                        <div class="temp">
+                        <small>
+                            <small>TEMP:</small>
+                        </small>
+                        {{weather.temperature}}&deg;
+                        <small>C</small>
+                        </div>
+                        <div class="wind">
+                        <small>
+                            <small>WIND:</small>
+                        </small>
+                        {{weather.windSpeed}} km/h
+                        </div>
+                        <div class="humidity">
+                        <small>
+                            <small>HUMIDITY:</small>
+                        </small>
+                        {{weather.humidity}}%
+                        </div>
+                        <div>&nbsp;</div>
+                    </div>
+                    <div class="icon">
+                        <div :class=weather.weatherDescription></div>
+                    </div>
+                    </div>
+                    <div class="future">
+                    <div class="day" :class="{ 'current-weather': index === 0 }">
+                        <h4 class="date">{{weather.date}}</h4>
+                    </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+            <p class="weather-history-label">Weather History</p>
```

Comment:
This feels like a separate component. Can it be extracted to keep things separate?